Umbraco 15/16 compatability

gavinfaux opened the pull request:
Remaining upgrade tasks:


- [x] Blog Import
- [x] Back Office Client
- [x] RSS Feed - Fix for /rss (posts) route
- [x] MetaWeblog / Live Writer support
- [x] Theme QA
  - [x] Vapour
  - [x] Material
  - [x] Mini
  - [x] Phantom
- [x] NuGet Package & Installation
- [x] Documentation (version/release notes for readme's, code/config comments)

Outstanding issues:

- [x] Mobile Markdown editor -wip, possibly seperate PR.

gavinfaux commented:
I think this is ready for review, will try and address markdown editor/live writer support next.

If I get stuck these outstanding problems could become issues for further work since they have the media issue in common and need a little more work to bring them upto date for 15+.  Would be nice to get this completed before v15 eol :wink:

Have thourghly tested in v15 and similar testing in v16. 

Have not bumped `@umbraco-cms/backoffice` to v16 yet, again can be reviewed later, likely compatible with v15 backoffice.

Sorry about the formatting changes it just makes things a little easier when doing diffs, though it's respecting .editorconfig so hopefully ok.

Another one for a bit later, some of these changes/fixes can be back ported to 13 LTS.

gavinfaux commented:
Ok it's not quite ready forget to remove the custom build file and revert to buildTransative and correct .csproj file, so a little more testing to do and hopefully get the Live Writer and Markdown editor completed.

Shazwazza commented:
Awesome stuff @gavinfaux - I'm off on holidays for a week so will get to this as soon as I'm back. Thx again, this is so awesome

Shazwazza commented:
Hi @gavinfaux!

For this next amazing release I had a few thoughts on simplifying:

* We don't need to port over all of the themes, and maybe it makes more sense to use a newer theme if you have a favorite. I'm not even sure if the old Material theme is still maintained for example. The purpose for all of those was to showcase the different features that could be used in themes since each one displays some over others.  Let me know what you think.
* Articulate used strongly typed models (i.e. things based of an IMasterModel) which has led to confusion in the community in some cases. I built Articulate a long time ago when working at HQ to showcase the flexibility/extensibility of Umbraco with ASP.NET and went the strongly typed model route to show how route hijacking can be leveraged and all sorts of other things. Now, it would probably be much simpler to remove the strongly typed models and use extension methods (or similar) to render stuff. Remember, this was built a looong time ago when we didn't have nice fancy things in ASP.NET Core.
* The front-end editor was built to showcase that you could write blog posts from your phone, while also attaching images, etc... I doubt anyone actually uses it (apart from me) but I could be wrong. I do think its a nice to have thing, but maybe it doesn't need to exist - again, let me know what you think.

gavinfaux commented:
Hi @Shazwazza 

I *was* confident this PR was ready for review, as it works as expected in Development environments.

However, the Production build is failing to render the front end. The view engine is only searching the default locations (/Pages/Shared/{0}.cshtml, /Views/Shared/{0}.cshtml, etc.) and subsequently fails to find the theme views in App_Plugins/Articulate/ that are resolved by PathHelper.GetThemeViewPath(...).

I am sure this worked previously, so I will check for any recent regressions. If none are found, my immediate plan is to register the 
required view locations manually, likely using RazorViewOptions or a similar approach.

The root cause appears to be a disconnect between this package being a Microsoft.NET.Sdk class library and not a modern Microsoft.NET.Sdk.Razor project, which Umbraco and ASP.NET Core now promote. This is evident in the divergence from documented behaviour and the need for workarounds with tooling like MSBuild and NuGet.

To resolve this permanently, we should prioritise switching this project to a Razor Class Library (RCL). I will review the impact of doing this now, as it would address the current issue, align us with the .NET 'happy path', and preserve our future developer sanity.

Noted your other points, will respond in fuller once this blocker resolved.

gavinfaux commented:
Held up by a couple of issues presently, for a number of days... 😖 

1. CI/CLI builds - unable to produce NuGet package with required static web assets / build props; only Visual Studio Pack produces valid package; however may have a solution based on similar issue adapating the `MicrosoftAspNetCore.ClientAssets` technique used by the [Umbraco.AuthorizedServices](https://github.com/umbraco/Umbraco.AuthorizedServices/issues/62#issuecomment-3151045560) package.

2. An Umbraco Management API call to `DocumentTypeService.getItemDocumentTypeSearch( { query: 'Articulate',  skip: 0, take: 1, isElement: false, } )` fails only when NuGet package installed in a site (but works in dev env), url is sent without the query string. A simple reproduction package with the same call works fine, so a bit stumped on this one; same package references, same build options. 

  - Failing code traced to the `hey-api\client-fetch` package spreading the sent data object with it's client options object;  stepping through a working version never seems to load this dependency in the browser 😕 

  - As a workaround we could use a custom API call to get the Articulate guid (had that working before switching to Umbraco API) and then come back to this.

gavinfaux commented:
There were many conflicting targets, race conditons and packing concerns to resolve for a single RCL assembly to get the CI/CLI build and pack working. I've borrowed from Umbraco's approach to dependencies and build, and split up as follows:

  - `Articulate.csproj`
    - Main project as before, somewhat refactored, resolved most use of API's marked obsolete.
    - Routing, services and front end controllers + models,  view location expander and themes resolved via services.
    - Default themes and assets
    - Embedded Packaging and themes (for copy theme feature)
  - `Articulate.Api.Management.csproj`
    - Management API
    - Backoffice Extension
  - `Articulate.Web.csproj`
    - Markdown Editor + assets as has a dependency on Api to pass routes to view
    - Themes and related web/routing could/should move here however working as is for now.
    - TBA: Front end SPA to finish as switched to the CSP compliant version of Alpine.js

I've resolved and tested the issues I was having with release builds of host website installed via NuGet package('s now).

May have to clean up a little and will need to test more thoroughly.

Not tested with v16 as yet, possible need to maintain seperate targets for v16 onwards due to dependencies though that should be easier to do now...

Unsure how you would like PR now as many changes and not so easy to review diffs, any preferences  or all at once okay? Will reset PR branch off your last commit.

Minor CI build issue to fix due to new structure, will resolve shortly.

Shazwazza commented:
Hey @gavinfaux ! As of now you are the absolute master and legend of all things articulate :) I'm more than happy to take all your advice and work as-is. I can do some reviews for sure but will mostly be skimming through. I think its better to try to get something working and then shipped even if there's a boat load of changes.

Let me know how you'd like to proceed, maybe just let me know when you think its ready for review and testing and we can get this merged in and then iterate from there?

gavinfaux commented:
Build fixed in the mess that is my dev branch, so confident PR will be okay, have a few comments and such to clean up, plus change/dev docs and perhaps wiki updates after.

Breifly tested v16, front end and back office and seems fine.

My query with v16 + is more to do with managing `@umbraco-cms\backoffice` versions but will post in forums to see what consensus is.

May look at moving Theme views/assets to Web as currently using bit of a workaround to get the assets packed as contentFiles, idk why staticwebassests refuses to work with `dotnet pack` with this but VS Pack will just do it.

See enough msbuild binlogs to last a life time 🙏

gavinfaux committed:
Update Git ignore patterns

Add attributes file

gavinfaux committed:
Create global.json

**IMPORTANT** Using `"rollForward": "latestFeature" will prevent the build completing with an exception in the  `Microsoft.NET.Sdk.StaticWebAssets` task.

Reference: [Intermittent Error on Build](https://github.com/dotnet/aspnetcore/issues/57147)

gavinfaux committed:
Remove legacy App_Plugins\Articulate folder

- Angular based BackOffice extensions are replaced by the new Lit and TypeScript based BackOffice extensions in the `Articulate.Api.Management` Client project.
- Themes have moved to the `wwwroot\Views\Articulate\_System\` folder pending migration from ASP.NET Web Forms (Master Pages) to Layouts in ASP.NET Core and the ``Articulate.Web` project
- The mobile optimised Markdown Editor has moved to the `Articulate.Web` project - this is disabled on the frontend for now pending OpenIddict based authentication with the BackOffice Management API.

gavinfaux committed:
Updated Razor Themes

- global using's replaced by `_ViewImports.cshtml` per theme
- Latest supported 3rd party CSS and JavaScript assets, removed jQuery, latest Font Awesome free for branding (e.g. X)
- Accessibility and web vitals improvements
- Latest Google Tags/Analytics support
- Minor feature and bug fixes e.g. Author RSS links corrected, https links, random tag order

gavinfaux committed:
Remove legacy BackOffice API controllers/models

- Replaced by `Articulate.Api.Management` project

gavinfaux committed:
Refactor Articulate Themes, Views and asset services

- `ArticulateViewLocationExpander` resolves location of Articulate Views and Partials (not assets) for User themes, System themes and other Views (e.g. Markdown Editor); current active theme resolved by `ArticulateThemeResolver` service.
- `PathHelper` removed and replaced by View expander, amends to `DefaultThemes` to support asset resolution , extending `ArticulateConstants` with Path locations and assist with (my) typo's
- Theme Repository service to support BackOffice copy theme feature and listing available themes.
- Views/Partial are now compiled at build time into assembly
- Themes are also now embedded as resources in assembly to support BackOffice copy theme feature

- NOTE: Developer edit and browser refresh (hot reload) currently not implemented, Razor runtime compilation for improved DX pending/future PR: [Enable runtime compilation for a Razor class library] (https://learn.microsoft.com/en-us/aspnet/core/mvc/views/view-compilation?view=aspnetcore-9.0&tabs=visual-studio#enable-runtime-compilation-for-a-razor-class-library)

gavinfaux committed:
Update extension methods

- `ContentUrls` not used
- `UrlHelperExtensions` no longer used, ThemedAsset superseded and other methods moved to PublishedContentExtensions as IMasterModel extensions, IUrlHelper not used (UmbracoHelperExtensions simiraly
- Moved to Extensions namespace (better implicit using's support)
- Legacy Google Analytics UA extension method updated to uses Google Tags
- Migrate obsolete methods, e.g. ILocalizationService  -> ILanguageService , Parent/Children -> Parent()/Children()
- Image Crop enhancement to check for (legacy) properties with just a string path

gavinfaux committed:
Frontend Controller/Model updates

- Removed unused IVariationContextAccessor
- Basic nullable enhancements where possible
- Removed PathHelper calls, View Expander now finds
- Minor amends to update obsolete methods, use Async where possible,

gavinfaux committed:
Component/Composer updates

- Retire SendingContentHandler, logic moved to ContentSavingHandler
- Retire ServerVariablesParsingHandler, replace by Management API / Client
- Update ContentTypeSavingHandler  to set ListView property
- Update obsolete methods, basic nullable amends
-

gavinfaux committed:
BlogML import/export

- Update obsolete methods
- basic nullable support
- Import first image correctly as MediaItem, not just a path
- Enhance async usage

gavinfaux committed:
ArticulateMetaWeblogProvider  update

- update obsolete methods
- Fix first image import as MediaItem
- enhance async support

gavinfaux committed:
Property editor amends

- Simplify as BackOffice Client project implements most property editor features

gavinfaux committed:
Routing amends

- update obsolete methods (update to use IPublishedContentTypeCache and IDocumentCacheService)
- basic nullable support
- disabled MarkdownEditorRoute pending completion of updates (authentication and template updates)

gavinfaux committed:
Remain Services updates

- BackOfficeAuthService is just used by the Material theme TitleTile partial
- basic nullable support / namespace amends

gavinfaux committed:
RSS updates

- RssResult moved to Controllers
- GeneratedRegex
- basic nullable support

gavinfaux committed:
Packaging update

- Data type (theme picker, markdown editor, tags/categories, image picker) and associated doc type updates (+ Google Tag update)

gavinfaux committed:
Unused

- MarkdownEditorInitModel moved to .Web project with Controller/View
- file rename to IImageModel
- ReflectionHelper unused

gavinfaux committed:
Articulate project and build updates

- Razor class library
- update Solution project type guids to C# projects from ASP.NET MVC 4
- retire Articulate.targets
- update readme's
- Follow Umbraco CMS practices (Directory.Build.props) for project dependency managment, packaging and code format
- Work around to pack static assets as contentFiles; VS Pack, msbuild or dotnet pack refuse to pack as staticwebassets?!
- bump version.json to 6.0.0
- remove out of date migration guides for back office

gavinfaux committed:
BackOffice Management API

gavinfaux committed:
BackOffice Client extension source code

gavinfaux committed:
Articulate.Web project

- Just 'disabled' Markdown Editor currently
- **NOTE** This project packs as `Articulate` since it has dependencies on the API and original Articulate project (which now packs as `Articulate.Core`

gavinfaux committed:
Test website

- Updated to Umbraco CMS 15.4.4
- **NOTE** launches on https://localhost:44366 or http://localhost:44169 by default, amend in launchsettings.json and appsettings.json as required

gavinfaux committed:
Build script updates

gavinfaux committed:
missing favicon and appsettings-schema

gavinfaux committed:
Delete Articulate.Tests.Website.sln

gavinfaux commented:
Just tested this in a clean site and 2 issues to resolve:

1. Installing Articulate.nupkg is not automatically bringing in the other Articulate dependencies, will review, may need a meta package.
2. Smidge is not generating asset bundles, again will review....

Otherwise all there.

EDIT: 
1. Resolved, I'm a silly, not an issue, though build tweaked a little to not include development references.
2. Resolved, removed Smidge since Umbraco 16.1.1 requires `Umbraco.Community.Smidge` and did not want to have seperate branches/builds for 15 vs 16 (yet). Linking to static assets the old fashioned way. 
  - All Views and static assets moved to Web project where `dotnet pack` produces needed staticwebassets.
  - Articulate embeds views/assets from Web project for theme duplication feature.
 
3. Final bug to resolve, back office node picker not resolving picked node, similar issue to not finding Articulate doc type, will fix apply similar fix.

Hope to push up these changes shortly...

gavinfaux committed:
Remove Smidge

- No need for separate v16 build referencing Umbraco.Community.Smidge package
- Remove DefaultThemes class, replace with const
- Update Views to link to css/js directly
- Remove Component used for bundling

gavinfaux committed:
Remove legacy Grid editor views

gavinfaux committed:
Build fixes

Remove direct reference to transitive dependencies

gavinfaux committed:
Move Themes to Web project

- Resolve static web assets issue with `dotnet pack` and frontend finding assets to load
- Removed Smidge, load link to assets the usual way
- Articulate embeds Theme Views and assets from Web project

gavinfaux committed:
Fix issue with hey-api/client-fetch not resolving Umbraco API calls

Call using browser fetch api

gavinfaux committed:
Don't pack developer only package references, add PrivateAssets="All"

gavinfaux committed:
Move builder call to before  builder.Build()

Umbraco 16 'hybrid' release/production fails to resolve static assets if added after Build() call

gavinfaux commented:
Completed, ready for review, perhaps release?

Markdown Editor to resolve, tricky authenticating with BackOffice from frontend other than API Users which is not particually login friendly.

gavinfaux committed:
Remove/Resolve InternalsVisibleTo

Shazwazza commented:
Thanks @gavinfaux :) Simple question - for changing many of the internal contracts to public, is this necessary? I just want to prevent publicly available APIs when they aren't needed to allow for easier iterations and less breaking changes.

Shazwazza commented on Directory.Build.props:
```diff
@@ -1,9 +1,65 @@
+		<Content Include="$(MSBuildThisFileDirectory)icon.png" Pack="true" PackagePath="" Visible="false" />
+	</ItemGroup>
+	<PropertyGroup>
+		<Authors>Shannon Deminick</Authors>
```
@gavinfaux feel free to add your name :)

Shazwazza replied:
@gavinfaux ping on this

Shazwazza commented on LICENSE:
```diff
@@ -1,6 +1,6 @@
 The MIT License (MIT)
 
-Copyright (c) 2014 Shandem
+Copyright (c) 2025 Shandem
```
```suggestion
Copyright (c) 2025 Shazwazza
```

Shazwazza committed:
Update LICENSE

gavinfaux committed:
Revert 0256bfc - Remove/Resolve InternalsVisibleTo

gavinfaux committed:
Update Articulate.Api.Management.csproj

Fix PackageId typo

gavinfaux committed:
Move Views back to App_Plugins/Articulate

Safe to do now packaging issues resolved, this was an earlier workaround to prevent conflicts with API/Client build conflicts.

gavinfaux committed:
Copy XML docs to output dir on package installation

gavinfaux commented:
> Thanks @gavinfaux :) Simple question - for changing many of the internal contracts to public, is this necessary? I just want to prevent publicly available APIs when they aren't needed to allow for easier iterations and less breaking changes.

Commit fd9a31a Remove/Resolve InternalsVisibleTo maybe made it look worse than it is, I did think about not exposing API's that were not really needed; dependent projects needed access but have reverted change and back to using `[InternalsVisibleTo]`. 

I may have been over zealous but I don't really see anything that could be in use publically? Let me know if any need amending?

#### Audited Diffs against v5/v13 branches against PR

| Before  | After |
| ------------- | ------------- |
| Components\ConfigureArticulateMvcOptions.cs(7):internal class | Options\ConfigureArticulateMvcOptions.cs(8):internal class |
| ContentUrls.cs(28):internal HashSet<string> GetContentUrls | deleted |
| Controllers\RsdController.cs(63):internal class XmlResult  | Controllers\XmlResult.cs(8):internal |
| Controllers\UrlHelperExtensions.cs(8):internal static class | deleted |
| Controllers\UrlHelperExtensions.cs(10):internal static string | deleted |
| ImportExport\DisqusImporter.cs(19):// internal class | deleted |
| ReflectionHelper.cs(8):internal static class | deleted |
| Routing\ArticulateDynamicRouteSelectorPolicy.cs(26):internal class | internal |
| Routing\ArticulateFrontEndFilterConvention.cs(5):internal class | internal |
| Routing\ArticulateRouteTemplate.cs(6):internal struct | internal  |
| Routing\ArticulateRouteValueTransformer.cs(24):internal class | internal |
| Routing\RouteCacheRefresherFilter.cs(17):internal class | internal  |
| Routing\RouteCollectionExtensions.cs(14):internal static string RoutePathFromNodeUrl | internal |
| Services\ArticulateTagRepository.cs(21):internal class | internal |
| UmbracoHelperExtensions.cs(183):internal static IEnumerable<IPublishedContent> GetContentByAuthor | internal |
| ArticulateDynamicRouteAttribute.cs(4):public sealed class | Attributes\ArticulateDynamicRouteAttribute.cs(4):public sealed class - now internal + InternalsVisibleTo |
| Syndication\RssResult.cs(4):public class | Controllers\RssResult.cs(12):internal class |
| ContentExtensions.cs(15):public static class | Extensions\ContentExtensions.cs(9):public static class - now internal + InternalsVisibleTo |
| ImportExport\BlogMlImporter.cs(93):public int GetPostCount | ImportExport\BlogMlImporter.cs(88):public int GetPostCount - now + InternalsVisibleTo |
ImportExport\BlogMlImporter.cs(103):public async Task<bool> Import | ImportExport\BlogMlImporter.cs(98):public async Task<ImportResponseDto> Import - now internal + InternalsVisibleTo |
| Routing\ArticulateRootNodeCache.cs(11):public class | Routing\ArticulateRootNodeCache.cs(10):internal class |
| Routing\ArticulateRouter.cs(19):public class | Routing\ArticulateRouter.cs(17):internal class
| new | Services\ArticulateThemeRepository.cs(12):internal class |
| new | Services\ArticulateThemeResolver.cs(8):internal class |
| moved from TitleTile.cshtml in Material Theme | Services\BackOfficeAuthService.cs(9):internal class
| Syndication\RssFeedGenerator.cs(14):public class | Syndication\RssFeedGenerator.cs(14):internal class 
| MetaWeblog\ArticulateMetaWeb.cs:(45,46) private vars | MetaWeblog\ArticulateMetaWeblogRegexes.cs(6):internal static partial class | 
| Syndication\RssFeedGenerator.cs:(27,28) private vars | Syndication\RssFeedGeneratorRegexes.cs(6):internal static partial class | 
| Controllers\MardownEditorApiController.cs(194): private var | API project: internal static partial class |
| n/a | API project: additions x 4 internal class |

gavinfaux commented:
I've not tested an 'upgrade' from v5 to v6 - I'll look at testing an older project if I still have one around (likely no media but should suffice).

I moved Views back to App_Plugins the previous location was a workaround for a conflict/packing issue with BackOffice.

gavinfaux commented:
Theme Picker loses it's editor on update from v5 to v6, reviewing.

Of the internal changes , perhaps `ContentExtension` could  be in use publically for custom imports, though same can be achieved with Content API.

Shazwazza commented:
The internals stuff isn't critical, more just reducing surface area of supported public apis, happy to take your lead there. For upgrades, i wouldn't worry too much about that at all. Upgrades for previous major updates are basically - export to blobml and reimport.

readingdancer commented:
Great work on this @gavinfaux that's one mammoth PR 🚀

gavinfaux committed:
Articulate Property Editors

- Full clone of src/Umbraco.Web.UI.Client/src/packages/markdown-editor
- Prevent conflicts if both Markdown Editors on content type
- Resolve Markdown Editor and Theme Picker property type alias to Backoffice editor element
- TODO: Remove Markdown Editor  when the Markdig package is merged into Umbraco
- See: https://github.com/umbraco/Umbraco-CMS/pull/19500

gavinfaux committed:
Update document-node-utils.ts

Umbraco API fetch clients

gavinfaux committed:
Update Articulate Management Api & Client

- Resolve issue with Articulate.MarkdownEditor and Umbraco.Markdown on same content/composition
- Resolve Property Editor aliases to match v5
- Articulate.MarkdownEditor: full(er) clone of Umbraco Web UI Client version; resolve clashes with default
- use Umbraco.Markdown schema for forward compatibility when migrating Articulate.MarkdownEditor
- use default Umbraco.Markdown prev values for  Articulate.MarkdownEditor
- regenerate client api

gavinfaux committed:
Articulate.Api.Management XML doc update

- Amends for removal of Articulate.Extension namespace
- Add Model to implicit usings

gavinfaux committed:
Remove Externsions namespace

- Avoid breaking v5 Public Api's
- Mark Obsolete methods and implement
- Review internals
- Review Async
- Update package with latest Data Types
- Map Articulate Richtext Editor to TipTap in v6.0 (so CMS v15.4.4 onwards) to ensure migration > 16.1.1

gavinfaux committed:
View updates after Articulate.Extensions revert

gavinfaux committed:
Minor fomatting

gavinfaux committed:
v5 API

- Restore v5 API, mark as obsolete with reference to v6 API

gavinfaux committed:
Static bundles for assests

gavinfaux committed:
Update ArticulateSwaggerOptions.cs

bugfix: Move warning to else block

gavinfaux committed:
Move ArticulateOperationIdRegexes

move to Articulate.Api.Management.Swagger folder/namespace

gavinfaux committed:
Update input-markdown.element.ts

Amend default editor heading to H2 as H1 is post title

gavinfaux committed:
Standardize use of GetCropUrl

-  Use Umbraco.Extensions.FriendlyImageCropperTemplateExtensions for crops
- Remove unused custom crop extensions
- Mark GetArticulateCropUrl as obsolete

gavinfaux committed:
Update PublishedContentExtensions.cs

InRandomOrder can be public

gavinfaux committed:
Update .gitignore

ignore user custom theme folder in test site

gavinfaux committed:
Client package updates

gavinfaux committed:
Update appsettings.json

remove sqlite umbracoDbDSN connection string, user should specify connection string on install of test site

gavinfaux committed:
Update Articulate.csproj

Bug fix: Correct path to embedded theme source files

gavinfaux committed:
Update PublishedContentExtensions.cs

just formatting

gavinfaux committed:
Package installation and migrations

- Ensure Articulate data types match previous alias values and use V15+ compatible property editor schemas
- Migrate Articulate Richtext data type to TipTap (runs after Core RTE migrations)

gavinfaux commented:
Still testing this and release notes... hope to complete soon.

gavinfaux committed:
Articulate Management API review

- Remove TagActionsBy call in SwaggerOptions, Umbraco already handles, this was causing Swagger API endpoint to merge all schemas when Delivery API enabled
- add ManagementApiRoute to customise API route from `/umbraco/management/{version}/articulate/{template}`  to `/umbraco/articulate/{version}/{template}`
- Review ArticulateOperationIdHandler to customise how client API is generated (formatting, clean and dx friendly)

gavinfaux committed:
Client API update

- Downgrade `@hey-api/openapi-ts` to version currently supported by `umbraco-extension' projects
- Add last supported legacy  `@hey-api/client-fetch` package, unbundled. Latest ` @hey-api/openapi-ts` can be updated in v16.3+ see https://github.com/umbraco/Umbraco-CMS/issues/19893
- Regenerate Client to reflect API/Swagger/generated code changes

gavinfaux committed:
Enable Delivery API in test site

- Testing impact on Umbraco Delivery and Management API Swagger generation

gavinfaux committed:
Update ArticulatePlanComponent.cs

Remove unused IRuntimeState parameter

gavinfaux committed:
v6 Release notes and minor readme update

Shazwazza commented:
@gavinfaux when you are ready let me know, I'll pull down a clean checkout of this branch and test locally. Hopefully I can export my current blog and re-import as a good test of functionality.

gavinfaux commented:
Had a last minute issue with Swagger when enabling the Delivery API, all resolved now.

I'm happy with this now, tested across v15.4 to v16.2.0 RC2, also tested upgrading from 10 -> 13 -> 15 -> 16. 

Let me know if anything else needed, changes required or need discussion.

And yes, import/export should work fine for getting around any 'upgrade' issues (unsure if it exports any extra media that has been written to the /media/articulate folder and does not have an entry in the Media library though).

Correction: A migration is run to change RTE editor from TinyMCE to TipTap, doing it in a migration rather than package.zip so that the prerequisite Umbraco TinyMCE migrations run first and then the data type change.

gavinfaux committed:
Make RssFeedGenerator public and update method visibility

Restore RssFeedGenerator from internal sealed to public class and updated method visibilities from private to protected virtual to allow for override/custom RssFeedGenerator. Also updated IRssFeedGenerator interface to use public method signature for GetFeed.

gavinfaux committed:
Rename MigrateV16DataTypeConfiguration to MigrateArticulateRichText

gavinfaux committed:
Save packaged media as ImageCropperValue JSON, not just a path

Fixes issue with having to to save Articulate Image Picker data type

gavinfaux committed:
Update README for v13 manual date type fix

This issue only applies to v13 now, fixed going forward

gavinfaux committed:
v5 User Theme View path compatibility

gavinfaux commented:
Reviewing for v17, extension fine, some API changes for .net10 .

Shazwazza commented:
@gavinfaux shall we just merge this in and continue working on it from there? I can take a 'quick' look over these changes to see if I can spot anything but think its prob best to just get this merged and iterate?

Shazwazza reviewed (approved):
Lets get this merged, there are far too many files to possibly review ;)

Shazwazza commented:
@gavinfaux this is AWESOME!!!!!!!!!!

Clone, run, everything just works - not sure if I was supposed to do anything else but things seem to be in place.

I exported my blog and re-imported it and it mostly worked. I used the base64 image option and most work but some don't. This could be the export engine, or something else. Not entirely sure at the moment but wow, this is exceptional work :)

From here, I think we look to get a beta release out soon?

Shazwazza commented:
Yep, looks like the exporter on the current prod version has issues when exporting with images, I get a ton of these

> System.UriFormatException: Invalid URI: The format of the URI could not be determined.
   at System.Uri.CreateThis(String uri, Boolean dontEscape, UriKind uriKind, UriCreationOptions& creationOptions)
   at System.Uri..ctor(String uriString)
   at Articulate.Syndication.RssFeedGenerator.GetBlogImage(IMasterModel rootPageModel) in /_/src/Articulate/Syndication/RssFeedGenerator.cs:line 127

> System.FormatException: String "[{"key":"8ccb63f8-8b3d-4ff7-a492-4071b8393cf0","mediaKey":"e61b15fb-c8b3-4131-ad22-d0e731f12bec"}]" is not a valid udi.
   at Umbraco.Cms.Core.UdiParser.ParseInternal(String s, Boolean tryParse, Boolean knownTypes, Udi& udi)
   at Umbraco.Cms.Core.UdiParser.Parse(String s)
   at Articulate.ImportExport.BlogMlExporter.AddBlogPosts(IContent archiveNode, BlogMLDocument blogMlDoc, String categoryGroup, String tagGroup, Boolean exportImagesAsBase64) in /_/src/Articulate/ImportExport/BlogMlExporter.cs:line 243

The import seems ok, so might need to see what I have to fix in the current version to get export and import working seamlessly.

gavinfaux commented:
Ported fix for media-udi export issue for 10-13 , will push PR soon.
Will base future PR form feature/umb15, Markdown Editor implemented + ASP.NET Core theme in progress butr eviewing v17 API changes  first.
